### PR TITLE
Fix #18962 - Moving columns with ENUM() datatype and DEFAULT value causes invalid SQL

### LINKED ISSUE #18962: Move columns with ENUM() & DEFAULT causes invalid SQL
### Describe the bug

Move columns fails to generate a valid `ALTER TABLE` statement when used on an `ENUM` field with a default value.

### To Reproduce

```sql
CREATE TABLE `test` (
  `id` int(11) NOT NULL,
  `enum` enum('yes','no') NOT NULL DEFAULT 'no',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci
```
1. Go to "Structure"
2. Click on "Move columns"
3. Drag `enum` above `id`
4. Click Go
5. Error: 
>  \# 1067 - Invalid default value for 'enum'

```sql
ALTER TABLE `test` 
CHANGE `enum` `enum` ENUM('yes','no') 
CHARACTER SET utf8 COLLATE utf8_unicode_ci NOT NULL 
DEFAULT '\'no\'' FIRST
```
Note the extraneous double quoting of the `DEFAULT 'no'`

### Expected behavior

The move columns feature should produce valid SQL for ENUM fields with a default value.

### Screenshots

image

image

### Server configuration

- Operating system: Rocky 8
- Web server: Apache
- Database version:  10.4.32-MariaDB-
- PHP version: 8.1.26
- phpMyAdmin version: 5.2.1-1

### PR DESCRIPTION
**Pull request by johnson361**
### Description

```
 SELECT
    `COLUMN_NAME` AS `Field`,
    `COLUMN_TYPE` AS `Type`,
    `COLLATION_NAME` AS `Collation`,
    `IS_NULLABLE` AS `Null`,
    `COLUMN_KEY` AS `Key`,
    `COLUMN_DEFAULT` AS `Default`,
    `EXTRA` AS `Extra`,
    `PRIVILEGES` AS `Privileges`,
    `COLUMN_COMMENT` AS `Comment`
FROM
    `information_schema`.`COLUMNS`  where `TABLE_SCHEMA` = 'test'  AND `TABLE_NAME` = 'test' ;;
```

When executing this query on MariaDB, the COLUMN_DEFAULT column returns default values enclosed in single quotes on both sides, causing an issue. However, in MySQL, this behavior is not observed. To address the inconsistency, I needed to trim off any extra single quotes that might exist in the COLUMN_DEFAULT values using TRIM('\\'' FROM COLUMN_DEFAULT`). Now, the issue is resolved.

Fixes #18962

You can see how the data appeared before the query modifications in the image below.
[
Issue
]

### CODE REVIEW

**Comment by johnson361** [comment]
@ChrisHSandN i have updated the logic . Please retest. Thank you .

File: yarn.lock [hunk 1]
```diff
@@ -5321,4 +5321,4 @@ yocto-queue@^0.1.0:
 zwitch@^1.0.0:
   version "1.0.5"
   resolved "https://registry.yarnpkg.com/zwitch/-/zwitch-1.0.5.tgz#d11d7381ffed16b742f6af7b3f223d5cd9fe9920"
-  integrity sha512-V50KMwwzqJV0NpZIZFwfOD5/lyny3WlSzRiXgA0G7VUnRlqttta1L6UQIHzd6EuBY/cHGfwTIck7w1yH6Q5zUw==
+  integrity sha512-V50KMwwzqJV0NpZIZFwfOD5/lyny3WlSzRiXgA0G7VUnRlqttta1L6UQIHzd6EuBY/cHGfwTIck7w1yH6Q5zUw==
```
Comment (kamil-tekiela) [inline yarn.lock:5324]: What's up with the newlines in unrelated files?

Comment (johnson361) [reply yarn.lock:5324]: Messed up the package.json and yarn.lock files .

File: libraries/classes/Controllers/Table/Structure/MoveColumnsController.php [hunk 2]
```diff
@@ -77,6 +77,16 @@ public function __invoke(): void
                 unset($data['Extra']);
             }
 
+             // If the database is MariaDB and the field type is enum, retrieve the default value using a different query to avoid additional quotes.
+             if ($this->dbi->isMariaDB() && $data['Field'] == 'enum') {
```
Comment (williamdes) [inline libraries/classes/Controllers/Table/Structure/MoveColumnsController.php:81]:
```suggestion
             if ($this->dbi->isMariaDB() && $data['Field'] === 'enum') {
```

**Comment by williamdes** [comment]
Closing this one since https://github.com/phpmyadmin/phpmyadmin/pull/19105 got merged

Thank you for contributing to fix this issue. Please test with the new changes and let us know
